package dmModelDDF;

import "ddf/ddf_extensions.proto";
import "ddf/ddf_math.proto";

option java_package = "com.dynamo.model.proto";
option java_outer_classname = "Model";

message ModelDesc
{
    required string mesh        = 2 [(resource)=true];
    required string material    = 3 [(resource)=true];
    repeated string textures    = 4 [(resource)=true];

    optional string name        = 10; // Deprecated
}

message SetConstant
{
    required uint64         name_hash   = 1;
    required dmMath.Vector4 value       = 2 [(field_align)=true];
}

message ResetConstant
{
    required uint64         name_hash   = 1;
}

message SetTexture
{
    required uint64 texture_hash = 1;
    required uint32 texture_unit = 2;
}
